api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -129,6 +128,10 @@ public UserServiceImpl(
     this.institutionService = institutionService;
   }
 
+  private int getCurrentDuccVersion() {
+    return configProvider.get().featureFlags.enableV3DataUserCodeOfConduct ? 3 : 2;
```

Comment:
Can we just keep the version directly in either the config file (simplest for now)? In the short/medium term, I'd really like to see a table in the database for the different singable compliance documents and their active & retired versions that each document view can refer to and user signature record can link to?

Reply:
We cannot, because it's feature flagged.

Reply:
We can add an integer to the config file without breaking things that don't know about it, right? Or are you saying that it exists in both states at once, guarded by a feature flag? In that case, we should remove it from the flag and just have a value for the active version.

In other words, we really shouldn't have code like this line. There's just no way it's going to age well.

Reply:
I'm saying that if I follow your suggestion, it will break the feature until the feature flag is turned on everywhere.

Reply:
Oh. You could use a feature flag for `enableDataUserCodeOfConductBeta`, and then have two values like `dataUserCodeOfConductBetaVersion` and `dataUserCodeOfConductCurrentVersion`. Then that line of code could keep working when one or both version keys changes.